agent/consul/multilimiter/multilimiter.go

```diff
@@ -145,14 +155,45 @@ func (m *MultiLimiter) Allow(e LimitedEntity) bool {
 			config = prefixConfig
 		}
 	}
-
 	limiter := &Limiter{limiter: rate.NewLimiter(config.Rate, config.Burst)}
-	limiter.lastAccess.Store(now)
-	tree, _, _ := limiters.Insert(e.Key(), limiter)
-	m.limiters.Store(tree)
+	limiter.lastAccess.Store(unixNow)
+	m.LimiterCh <- &LimiterWithKey{l: limiter, k: e.Key(), t: now}
```

Comment:
I think we mentioned it's preferable to drop limiters when the buffer is full rather than blocking. Shall we do a non-blocking send here?
```suggestion
	select {
	case m.LimiterCh <- &LimiterWithKey{l: limiter, k: e.Key(), t: now}:
	default:
		// Write buffer is full. We choose to drop limiters rather than block to prioritize throughput.
	}
```

Reply:
After discussion we decided that this no different from locking, so no need to drop the limiter